### ISSUE TITLE
BelongsTo with dependsOn not working as expected (visibillity)

### ISSUE THREAD

**Issue by RomkaLTU** [laravel/nova-issues#4792]
- Laravel Version: 9.29.0
- Nova Version: 4.13.0
- PHP Version: 8.1
- Database Driver & Version: Mysql 8.0.30
- Operating System and Version: Ubuntu 22 LTS
- Browser type and version: All

### Description:

I'm unable to toggle field visibility

```
BelongsTo::make('Sport', 'sport', Sport::class)
                ->hide()
                ->dependsOn(['type'], function (BelongsTo $field, NovaRequest $request, FormData $formData) {
                    if ($formData->type == '20') {
                        $field->show();
                    }
                }),
```

Should be hidden by default ant show when $formData->type == '20'
I'm only able to do this with ->readonly() / ->readonly(false)

**Comment by crynobone** [comment]
Unable to reproduce the issue, please provide full reproducing repository based on fresh installation as suggested in the bug report template (or you can refer to https://github.com/nova-issues for example).

We already have multiple dusk tests covering this use case and it would have pick this issue if it does exists. I'm not saying that the bug you are seeing if not true but we don't have enough information based on the description above to replicate it.

**Comment by RomkaLTU** [comment]
You're right just tried on a fresh Laravel install and it's working. Trying to track down a package that breaks this functionality.

**Comment by RomkaLTU** [comment]
This plugin partially breaks BelongsTo field: https://github.com/codenco-dev/nova-grid-system

**Comment by crynobone** [comment]
Should be solved via codenco-dev/nova-grid-system#23
